autorest/azure/azure.go

```diff
@@ -50,8 +51,8 @@ type RequestError struct {
 
 // Error returns a human-friendly error message from service error.
 func (e RequestError) Error() string {
-	return fmt.Sprintf("azure: Service returned an error. Code=%q Message=%q Status=%d",
-		e.ServiceError.Code, e.ServiceError.Message, e.StatusCode)
+	return fmt.Sprintf("autorest/azure: Service returned an error. %v Status=%v",
```

Comment:
@garimakhulbe as long as it looks fine it is good. that's why I also would love to see a real string generated from this method in a test so that we can see how it actually behaves in real life and design it from there. My imagination is not strong enough to picture what kind of a string this method will return.